framework/Metrics/DTW.py

```diff
@@ -76,53 +81,67 @@ def _localReadMoreXML(self,xmlNode):
     if not self.wrongKeywords:
       self.raiseAnError(IOError,'The DTW metrics block contains parameters that are not recognized: ' + str(self.wrongKeywords))
 
-  def distance(self,x,y):
+  def __evaluateLocal__(self, x, y, weights = None, axis = 0, **kwargs):
     """
-      This method set the data return the distance between two histories x and y
-      @ In, x, dict, dictionary containing data of x
-      @ In, y, dict, dictionary containing data of y
-      @ Out, value, float, distance between x and y
+      This method computes DTW distance between two inputs x and y based on given metric
+      @ In, x, numpy.ndarray, array containing data of x, if 1D array is provided,
+        the array will be reshaped via x.reshape(-1,1), shape (n_samples, ), if 2D
+        array is provided, shape (n_samples, n_time_steps)
+      @ In, y, numpy.ndarray, array containing data of y, if 1D array is provided,
+        the array will be reshaped via y.reshape(-1,1), shape (n_samples, ), if 2D
+        array is provided, shape (n_samples, n_time_steps)
+      @ In, weights, None or array_like (numpy.array or list), optional weights associated
+        with input, shape (n_samples) if axis = 0, otherwise shape (n_time_steps)
+      @ In, axis, integer, axis along which a metric is performed, default is 0,
+        i.e. the metric will performed along the first dimension (the "rows").
+        If metric postprocessor is used, the first dimension is the RAVEN_sample_ID,
+        and the second dimension is the pivotParameter if HistorySet is provided.
+      @ In, kwargs, dictionary of parameters characteristic of each metric
```

Comment:
@ In, kwargs, dict, dictionary of parameters characteristic of each metric

Reply:
fixed